{
  "metadata": {
    "toc-showmarkdowntxt": true,
    "toc-showcode": true,
    "toc-showtags": true,
    "language_info": {
      "codemirror_mode": {
        "name": "python",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.8"
    },
    "kernelspec": {
      "name": "python",
      "display_name": "Python (Pyodide)",
      "language": "python"
    }
  },
  "nbformat_minor": 4,
  "nbformat": 4,
  "cells": [
    {
      "cell_type": "markdown",
      "source": "Import the necessary Biopython libraries\nThe libraries make the work easier",
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": "from Bio import SeqIO\nfrom Bio.Blast.Applications import NcbiblastnCommandline\nfrom Bio.Blast.Applications import NcbimakeblastdbCommandline\nfrom Bio.Blast import NCBIWWW, NCBIXML\n",
      "metadata": {
        "trusted": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": "Open the original condensed cry genes file.\nThe file labelled 'crygene123' contains all the cry genes from Bacillus thrugiensis",
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": "#open the original condensed cry genes \nwith open ('cry123.fasta', 'r') as handle:\n    for record in SeqIO.parse(handle, 'fasta'):\n        biomaker_seq = record.seq\n",
      "metadata": {
        "trusted": true
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": "#open the maize genome as from ncbi\nwith open ('mz.fasta', 'r') as handle:\n    for record in SeqIO.parse(handle, 'fasta'):\n        maize_seq = record.seq\n",
      "metadata": {
        "trusted": true
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": "from Bio import Align\n\n# Define your sequences\nbiomarker = biomaker_seq\nmaize = maize_seq\n\n# Create a PairwiseAligner instance\naligner = Align.PairwiseAligner()\n\n# Perform local sequence alignment\nalignments = aligner.align(biomarker, maize)\n\n# Print the local alignment results\nfor alignment in alignments:\n    print(alignment)\n",
      "metadata": {},
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": "",
      "metadata": {},
      "execution_count": null,
      "outputs": []
    }
  ]
}